interfaces/builtin/hugepages_control.go

```diff
@@ -0,0 +1,48 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const hugepagesControlSummary = `allows controlling hugepages`
+
+const hugepagesControlBaseDeclarationSlots = `
+  hugepages-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const hugepagesControlConnectedPlugAppArmor = `
+# Description: Allow control to hugepages.
+/sys/kernel/mm/{hugepages,transparent_hugepage}/{,**} rw,
+/dev/hugepages/{,**} rwk,
+/run/hugepages/{,**} rwk,
+`
+
+func init() {
+	registerIface(&commonInterface{
+		name:                  "hugepages-control",
+		summary:               hugepagesControlSummary,
+		implicitOnCore:        true,
```

Comment:
What is providing /run/dpdk on the host on core? Are you expecting snaps to ship stuff that write to there? Would that not be properly implemented as a slot?